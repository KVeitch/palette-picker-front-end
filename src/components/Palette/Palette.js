import React from 'react';
import './Palette.scss'


function Palette(props) {
  const {palette_name, color0, color1, color2, color3, color4} = props.palette
  const colors = [color0, color1, color2, color3, color4]

  const displayColors = colors.map(color => {
    return(<div className="palette__color" style={{backgroundColor:`#${color}`}}> 
      </div>)
  }) 


  return (
    <div className="palette__container">
      <div className="palette__container--name">
        {palette_name}
      </div>
      {displayColors}
    </div>
  );
}

export default Palette;

