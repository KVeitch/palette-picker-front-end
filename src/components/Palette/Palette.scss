.palette__container{
  display:flex;
  width:300px;
  justify-content: space-evenly;
  border:solid 1px gray;
  padding:2px 5px
}
.palette__container--name{
  width:100px;
  text-align: left;
  line-height: 30px;
}
.palette__color{
  height:30px;
  width:30px;
}