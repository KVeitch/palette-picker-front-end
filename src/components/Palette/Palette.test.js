import React from 'react';
import { shallow } from 'enzyme';
import Palette from './Palette';

describe('Palette', () => {
  let wrapper;
  const mockPalette = {"id": 1,"palette_name": "Colors","project_id": 1,"color0": "6320ee","color1": "D5f2e3","color2": "999950","color3": "C7B446","color4": "C2B078"}
  beforeEach(() => {
    wrapper = shallow(<Palette key={1} palette={mockPalette}/>);
  });

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
