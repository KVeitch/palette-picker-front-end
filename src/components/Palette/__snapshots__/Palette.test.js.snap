// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="palette__container"
>
  <div
    className="palette__container--name"
  >
    Colors
  </div>
  <div
    className="palette__color"
    key="div0"
    style={
      Object {
        "backgroundColor": "#6320ee",
      }
    }
  />
  <div
    className="palette__color"
    key="div1"
    style={
      Object {
        "backgroundColor": "#D5f2e3",
      }
    }
  />
  <div
    className="palette__color"
    key="div2"
    style={
      Object {
        "backgroundColor": "#999950",
      }
    }
  />
  <div
    className="palette__color"
    key="div3"
    style={
      Object {
        "backgroundColor": "#C7B446",
      }
    }
  />
  <div
    className="palette__color"
    key="div4"
    style={
      Object {
        "backgroundColor": "#C2B078",
      }
    }
  />
</div>
`;
